www-apps/gitea: version bump + maintainership

nemunaire opened the pull request:
Hi!

Here is a version bump for `www-apps/gitea`.

Please find attached 3 commits:

* **taking maintainership:** As discussed [here](https://github.com/gentoo/gentoo/pull/10947), [here](https://github.com/gentoo/gentoo/pull/10601),  [here](https://github.com/nemunaire/gentoo-overlay/commit/663269936445ac8964bf4156eca361b71e6b158e#comments) and [here](https://github.com/nemunaire/gentoo-overlay/commit/54a8bc41f9d65c619d059d32ec05573bf4858718#commitcomment-32141171) with @xdch47, we will both co-maintain the ebuild.

* **bump to 1.7.1:** The latest version released is 1.7.1. It includes some cleaning regarding configuration files (`GITEA_WORK_DIR` is now used as working directory instead of `GITEA_CUSTOM`, see [gitea doc](https://docs.gitea.io/en-us/specific-variables/)). Also closes: https://bugs.gentoo.org/671596.

* **remove old:** Old Gitea releases contains security vulnerabilities, so we remove vulnerable ebuild from the tree + do some cleaning with old files. `gitea.logrotated` is not needed anymore as `gitea` include it's own logrotate mechanism. `app.ini` is not used, instead we edit the sample config through `sed` during the ebuild prepare step.

@hydrapolic could you review and give us your feedback?

hydrapolic commented on www-apps/gitea/gitea-1.7.1.ebuild:
```diff
@@ -31,21 +31,31 @@ pkg_setup() {
-	GOPATH="${WORKDIR}/${P}:$(get_golibdir_gopath)" emake -C src/${EGO_PN} generate
-	TAGS="bindata pam sqlite" LDFLAGS="" CGO_LDFLAGS="" GOPATH="${WORKDIR}/${P}:$(get_golibdir_gopath)" emake -C src/${EGO_PN} build
+	GOPATH="${WORKDIR}/${P}:$(get_golibdir_gopath)" emake -C "src/${EGO_PN}" generate
+	TAGS="bindata pam sqlite" LDFLAGS="" CGO_LDFLAGS="-fno-PIC" GOPATH="${WORKDIR}/${P}:$(get_golibdir_gopath)" emake -C "src/${EGO_PN}" build
```
Why -fno-PIC?

nemunaire replied:
It was there since the beginning (before the last bump). I wasn't be able to find any reason for its initial inclusion, nor for this removal in the last bump. Perhaps we should continue without it?

hydrapolic replied:
Unless there is a reason, I think we should not disable PIC.

hydrapolic commented on www-apps/gitea/gitea-1.7.1.ebuild:
```diff
@@ -31,21 +31,31 @@ pkg_setup() {
 	sed -i -e "s/\"main.Version.*$/\"main.Version=${PV}\"/"\
-		-e "s/-ldflags '-s/-ldflags '/" src/${EGO_PN}/Makefile || die
+		-e "s/-ldflags '-s/-ldflags '/" \
+		-e "s/GOFLAGS := -i -v/GOFLAGS := -v/" \
```
Why removing -i from GOFLAGS?

nemunaire replied:
The `-i` flag seems to be responsible for some cases of GO stdlib recompilation, causing permissions errors, I encountered. Described in this issue: https://github.com/golang/go/issues/24674#issuecomment-379085520

By the way, `-i` is convenient when developing, as it speed up futhers recompilations, there is no gain in our case as we only compile one time.

hydrapolic commented:
I'm glad you both are working together on this, will test asap, thanks!

hydrapolic commented on www-apps/gitea/gitea-1.7.1.ebuild:
```diff
@@ -0,0 +1,76 @@
+pkg_postinst() {
+	if [[ ! -e "${EROOT}/var/lib/gitea/conf/app.ini" ]]; then
+		elog "No app.ini found, copying initial config over"
+		cp "${FILESDIR}"/app.ini "${EROOT}"/var/lib/gitea/conf/ || die
```
Since app.ini is removed, this will fail.

nemunaire commented:
Added some USE flags and fixed `sqlite` that also requires `sqlite_unlock_notify`, as noted [here](https://docs.gitea.io/en-us/install-from-source/).

```diff
--- a/www-apps/gitea/gitea-1.7.1.ebuild
+++ b/www-apps/gitea/gitea-1.7.1.ebuild
@@ -13,6 +13,7 @@ SRC_URI="https://github.com/go-gitea/gitea/archive/v${PV}.tar.gz -> ${P}.tar.gz"

 LICENSE="MIT"
 SLOT="0"
+IUSE="sqlite pam"

 DEPEND="
        dev-go/go-bindata
@@ -20,7 +21,7 @@ DEPEND="
 "
 RDEPEND="
        dev-vcs/git
-       sys-libs/pam
+       pam? ( sys-libs/pam )
 "

 pkg_setup() {
@@ -46,7 +47,12 @@ src_prepare() {

 src_compile() {
        GOPATH="${WORKDIR}/${P}:$(get_golibdir_gopath)" emake -C "src/${EGO_PN}" generate
-       TAGS="bindata pam sqlite" LDFLAGS="" CGO_LDFLAGS="" GOPATH="${WORKDIR}/${P}:$(get_golibdir_gopath)" emake -C "src/${EGO_PN}" build
+       local my_tags=(
+               bindata
+               $(usex pam 'pam' '')
+               $(usex sqlite 'sqlite sqlite_unlock_notify' '')
+       )
+       TAGS="${my_tags[@]}" LDFLAGS="" CGO_LDFLAGS="" GOPATH="${WORKDIR}/${P}:$(get_golibdir_gopath)" emake -C "src/${EGO_PN}" build
 }

 src_install() {
```

hydrapolic commented on www-apps/gitea/gitea-1.7.1.ebuild:
```diff
@@ -8,19 +8,20 @@ EGO_PN="code.gitea.io/gitea"
 
 LICENSE="MIT"
 SLOT="0"
+IUSE="sqlite pam"
```
Please sort

hydrapolic replied:
Does USE=sqlite work without sqlite being installed?

Perlovka commented:
I would suggest to use gitea user instead of git. It's already used by gitolite.

Perlovka commented on www-apps/gitea/gitea-1.7.1.ebuild:
```diff
@@ -8,19 +8,20 @@ EGO_PN="code.gitea.io/gitea"
 
 DEPEND="
 	dev-go/go-bindata
 	sys-libs/pam
```
Double depend on pam?

nemunaire replied:
Yes: pam is required during the build (along with `dev-go/go-bindata`) and is also required at runtime, but not `dev-go/go-bindata`.

Perlovka commented:
Also, maybe rename gitea.initd-r1 to just gitea.initd?

nemunaire commented:
@xdch47 address your concerns. The ebuild has been updated!

hydrapolic commented on www-apps/gitea/gitea-1.7.1.ebuild:
```diff
@@ -0,0 +1,103 @@
+S="${WORKDIR}/${P}/src/${EGO_PN}"
+
+PATCHES=(
+	"${FILESDIR}/fix-chpw-loop.patch"
```
Is this accepted upstream? Why is it needed?

nemunaire replied:
Accepted upstream: https://github.com/go-gitea/gitea/commit/f4c7e87fc9c9ff0659a33affc0136b38da30e0aa in queue for the next release: 1.7.3

xdch47 commented:
yes, I patched and submitted it upstream (pending), since otherwise creating new-user results in an infinite loop for sub-url (see: https://github.com/go-gitea/gitea/pull/5965, https://github.com/go-gitea/gitea/issues/5815  and [gitea-forum](https://discourse.gitea.io/t/redirect-loop-for-initial-password-change-behind-nginx-ssl-reverse-proxy/684))

--> fix change password redirect loop on sub-urls

xdch47 commented:
@Perlovka I would prefer to keep the `git` username, because it seem odd to me to clone via a "gitea" - user 
Is it likely to have a lot of users installing gitolite and gitea both on the same time ? 
And shouldn't we let the user decide, which one to rename ? (I do have a `gitolite@...` and a `git@...`)

xdch47 commented:
i mean, my feeling is that git@*\<URL\>/\<USER or GROUP\>/\<REPO\>* is one category:
--> it implies kind of *github/gitlab/..* -style [http(s),(user-)configuration, issues, forks, wiki, etc] ...

whereas *gitolite* is for me a separate (more secure, but less features) category -- git@*\<URL\>:\<REPO\>*

Perlovka commented:
@xdch47 yes, it shoud use git user, as I stated in post scriptum ;)

nemunaire commented:
PR updated following 1.7.2 release.

xdch47 commented:
1.7.3 has been released - patch is upstream.
